omod/src/main/java/org/openmrs/module/owa/activator/OwaActivator.java

```diff
@@ -163,4 +172,32 @@ public void stopped() {
 		log.info("OWA Module stopped");
 	}
 	
+	/**
+	 * Used to set the login Url Global Property that directs to the Add-on OWA UI if no UI module
+	 * is installed. Searches against the legacy and Reference Application module on installed UI modules.
+	 */
+	private void configureLoginUrl(AppManager appManager) {
+		if (appManager.exists(ADD_ON_MANAGER_APP)) {
+			if (!uiModuleInstalled()) {
+				String folderName = null;
+				for (App app : appManager.getApps()) {
+					if (app.getName().equalsIgnoreCase(ADD_ON_MANAGER_APP)) {
+						folderName = app.getFolderName();
```

Comment:
Did you take a look at app.getLaunchUrl()?

Reply:
Thank you. I'd not internalized it. Including it into the changes.

Reply:
getLaunchUrl() depends on a baseUrl that is set by the user on the OWA module interface/view. Until this is set, then getLaunchUrl() returns null. But I've used it just in case the user changes/sets the baseUrl.